packages/axe-result-converter/src/combined-report-data-converter.ts

```diff
@@ -67,18 +65,13 @@ export class CombinedReportDataConverter {
         return failuresGroup.sort(this.compareFailureGroup);
     }
 
-    private getFailureData(results: AxeResult[]): FailureReportData {
-        let urlsCount = 0;
+    private getFailureData(results: AxeResult[]): FailureData[] {
         const failureData: FailureData[] = [];
         if (results) {
```

Comment:
The nested ifs and for loops make this kind of difficult to read. When would we get null/undefined for results/result/node?

Reply:
When data provided malformed for any reason.